src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -52,7 +52,55 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                 .span_label(span, format!("use of possibly uninitialized {}", item_msg))
                 .emit();
         } else {
-            let msg = ""; //FIXME: add "partially " or "collaterally "
+            // FIXME Should these be moved to impl Place?
```

Comment:
Where should I move these functions?

Reply:
Yes, you can move both this functions to be methods of [`Place`](https://github.com/rust-lang/rust/tree/master/src/librustc/mir/mod.rs).

Reply:
I think, if it were me, I would put make an "extension trait" and keep the methods here. They seem pretty tied to borrow check.